File: tests/algorithms/x/PHP/graphics/butterfly_pattern.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _append($arr, $x) {
    $arr[] = $x;
    return $arr;
}
$__start_mem = memory_get_usage();
$__start = _now();
  function repeat_char($ch, $count) {
  $result = '';
  $i = 0;
  while ($i < $count) {
  $result = $result . $ch;
  $i = $i + 1;
};
  return $result;
};
  function butterfly_pattern($n) {
  $lines = [];
  $i = 1;
  while ($i < $n) {
  $left = repeat_char('*', $i);
  $mid = repeat_char(' ', 2 * ($n - $i) - 1);
  $right = repeat_char('*', $i);
  $lines = _append($lines, $left . $mid . $right);
  $i = $i + 1;
};
  $lines = _append($lines, repeat_char('*', 2 * $n - 1));
  $j = $n - 1;
  while ($j > 0) {
  $left = repeat_char('*', $j);
  $mid = repeat_char(' ', 2 * ($n - $j) - 1);
  $right = repeat_char('*', $j);
  $lines = _append($lines, $left . $mid . $right);
  $j = $j - 1;
};
  $out = '';
  $k = 0;
  while ($k < count($lines)) {
  if ($k > 0) {
  $out = $out . '
';
}
  $out = $out . $lines[$k];
  $k = $k + 1;
};
  return $out;
};
  echo rtrim(butterfly_pattern(3)), PHP_EOL;
  echo rtrim(butterfly_pattern(5)), PHP_EOL;
$__end = _now();
$__end_mem = memory_get_peak_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;

File: tests/algorithms/transpiler/FS/graphics/butterfly_pattern.fs
// Generated 2025-08-13 16:13 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let _idx (arr:'a array) (i:int) : 'a =
    if not (obj.ReferenceEquals(arr, null)) && i >= 0 && i < arr.Length then arr.[i] else Unchecked.defaultof<'a>
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec repeat_char (ch: string) (count: int) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable ch = ch
    let mutable count = count
    try
        let mutable result: string = ""
        let mutable i: int = 0
        while i < count do
            result <- result + ch
            i <- i + 1
        __ret <- result
        raise Return
        __ret
    with
        | Return -> __ret
and butterfly_pattern (n: int) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable n = n
    try
        let mutable lines: string array = Array.empty<string>
        let mutable i: int = 1
        while i < n do
            let left: string = repeat_char ("*") (i)
            let mid: string = repeat_char (" ") ((2 * (n - i)) - 1)
            let right: string = repeat_char ("*") (i)
            lines <- Array.append lines [|((left + mid) + right)|]
            i <- i + 1
        lines <- Array.append lines [|(repeat_char ("*") ((2 * n) - 1))|]
        let mutable j: int = n - 1
        while j > 0 do
            let left: string = repeat_char ("*") (j)
            let mid: string = repeat_char (" ") ((2 * (n - j)) - 1)
            let right: string = repeat_char ("*") (j)
            lines <- Array.append lines [|((left + mid) + right)|]
            j <- j - 1
        let mutable out: string = ""
        let mutable k: int = 0
        while k < (Seq.length (lines)) do
            if k > 0 then
                out <- out + "\n"
            out <- out + (_idx lines (int k))
            k <- k + 1
        __ret <- out
        raise Return
        __ret
    with
        | Return -> __ret
ignore (printfn "%s" (butterfly_pattern (3)))
ignore (printfn "%s" (butterfly_pattern (5)))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
